src/vsphere_cpi/lib/cloud/vsphere/resources/cluster.rb

```diff
@@ -75,17 +77,35 @@ def vm_host_affinity_rule_name
       #
       # @return [String] Name of the host Group specified under a cluster config.
       def host_group
-        config.host_group
+        config.host_group_name
+      end
+
+      # Returns a host group rule if specified in cluster config. Otherwise,
+      # it returns a nil.
+      #
+      # @return [String] Name of the host Group rule type specified under a cluster config.
+      def host_group_drs_rule
+        config.host_group_drs_rule
       end
 
       # it returns a default name created by adding suffix to host group name
       #
       # @return [String] Name of the VM Group
       def vm_group
         return nil if host_group.nil?
-        return "vm_group-#{(0...6).map { (97 + rand(26)).chr }.join}" if host_group.length > 100
-        host_group + CLUSTER_VM_GROUP_SUFFIX
+        if host_group.length > 100
```

Comment:
I can't tell what this code is trying to do. I know the original was inscrutable as well. Can we fix it?